File: solutions/970.rb
# @param {Integer} x
# @param {Integer} y
# @param {Integer} bound
# @return {Integer[]}
def powerful_integers(x, y, bound)
  res = {}
  ibound = x == 1 ? 1 : Math.log(bound, x)
  jbound = y == 1 ? 1 : Math.log(bound, y)
  0.upto(ibound).each do |i|
    0.upto(jbound).each do |j|
      pn = x**i + y**j
      break if pn > bound

      res[pn] = nil
    end
  end

  res.keys
end

bound = 100
x = 1
y = 2

p powerful_integers(x, y, bound)

File: _site/solutions/970.py

from math import log


class Solution:
    def powerfulIntegers(self, x, y, bound):
        """
        :type x: int
        :type y: int
        :type bound: int
        :rtype: List[int]
        """
        ibound = 1 if x == 1 else int(log(bound, x))
        jbound = 1 if y == 1 else int(log(bound, y))
        res = set()
        for i in range(ibound + 1):
            for j in range(jbound + 1):
                tn = x ** i + y ** j
                if tn > bound:
                    break
                res.add(tn)

        return list(res)


x = 1
y = 2
bound = 100
print(Solution().powerfulIntegers(x, y, bound))
